automation/azure/azure_rbd_from_rg_tag/azure_rbd_from_rg_tag.pt

```diff
@@ -344,15 +322,155 @@ script "js_rbds", type: "javascript" do
     }
 
     if (rules.length > 0) {
-      result.push({
+      rule_lists.push({
         id: rbd_id,
         name: rbd_name,
-        verb: verb,
         effective_at: param_effective_date,
         rules: rules
       })
     }
   })
+
+  result = {
+    rule_lists: rule_lists,
+    dummy: "dummy"
+  }
+EOS
+end
+
+datasource "ds_existing_rbds" do
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, "flexera")
+    path join(["/finops-customizations/v1/orgs/", rs_org_id, "/rule-based-dimensions"])
+    # This is just to ensure that the new rules are generated before we gather existing RBDs
+    header "dummy", val($ds_new_rules, "dummy")
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "values[*]") do
+      field "id", jmes_path(col_item, "id")
+      field "name", jmes_path(col_item, "name")
+    end
+  end
+end
+
+datasource "ds_existing_rbds_that_match" do
+  run_script $js_existing_rbds_that_match, $ds_existing_rbds, $ds_new_rules
+end
+
+script "js_existing_rbds_that_match", type: "javascript" do
+  parameters "ds_existing_rbds", "ds_new_rules"
+  result "result"
+  code <<-'EOS'
+  new_ids = _.pluck(ds_new_rules["rule_lists"], "id")
+  result = _.filter(ds_existing_rbds, function(rbd) { return _.contains(new_ids, rbd["id"]) })
+EOS
+end
+
+datasource "ds_existing_rbds_with_rules" do
+  iterate $ds_existing_rbds_that_match
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, "flexera")
+    path join(["/finops-customizations/v1/orgs/", rs_org_id, "/rule-based-dimensions/", val(iter_item, "id"), "/rules/", $param_effective_date])
+    # We simply skip RBDs that don't have a ruleset with a conflicting date
+    ignore_status [ 400, 401, 403, 404, 429 ]
+  end
+  result do
+    encoding "json"
+    field "rules", jmes_path(response, "rules")
+    field "id", val(iter_item, "id")
+    field "name", val(iter_item, "name")
+  end
+end
+
+datasource "ds_rbds" do
+  run_script $js_rbds, $ds_new_rules, $ds_existing_rbds, $ds_existing_rbds_with_rules, $param_subscription_fallback
+end
+
+script "js_rbds", type: "javascript" do
+  parameters "ds_new_rules", "ds_existing_rbds", "ds_existing_rbds_with_rules", "param_subscription_fallback"
+  result "result"
+  code <<-'EOS'
+  existing_rbd_ids = _.pluck(ds_existing_rbds, "id")
+
+  // Create rules that mark the beginning and end of the generated AWS rules.
+  if (param_subscription_fallback == "Yes") {
+    start_block_value = "Start: Generated Azure Resource Group / Subscription Rules"
+    end_block_value = "End: Generated Azure Resource Group / Subscription Rules"
+  } else {
+    start_block_value = "Start: Generated Azure Resource Group Rules"
+    end_block_value = "End: Generated Azure Resource Group Rules"
```

Comment:
Since these are now stackable, it seems like we could remove the "subscription fallback" capability on the Azure RG PT ?